src/lib/agent/PolykeyAgent.ts

```diff
@@ -465,7 +553,105 @@ class PolykeyAgent {
       secretBuffer = Buffer.from(secretContent);
     }
     await vault.updateSecret(secretName, secretBuffer);
-    return UpdateSecretResponseMessage.encode({ successful: true }).finish();
+    return UpdateSecretResponseMessage.encodeDelimited({ successful: true }).finish();
+  }
+
+  /////////////////////
+  // Peer Operations //
+  /////////////////////
+  private async addPeer(nodePath: string, request: Uint8Array) {
+    const { publicKey, connectedAddress, relayPublicKey } = AddPeerRequestMessage.decodeDelimited(request);
+    const pk = this.getPolyKey(nodePath);
+    pk.peerManager.addPeer(new PeerInfo(publicKey, connectedAddress, relayPublicKey));
+    return AddPeerResponseMessage.encodeDelimited({ successful: true }).finish();
+  }
+  private async getPeerInfo(nodePath: string, request: Uint8Array) {
```

Comment:
Are CLI commands being mapped to these methods? Ideally the GRPC interface and CLI interface should be sharing code where possible.